python/packages/autogen-agentchat/src/autogen_agentchat/ui/_console.py

```diff
@@ -22,11 +26,53 @@ def _is_output_a_tty() -> bool:
 T = TypeVar("T", bound=TaskResult | Response)
 
 
+class UserInputManager:
+    def __init__(self, callback: UserProxyAgent.InputFuncType = UserProxyAgent.DEFAULT_INPUT_FUNC):
```

Comment:
perhaps the `InputFuncType` should be outside of the `UserProxyAgent` in a common place, e.g., `autogen_agentchat.base` or something like that? It's a bit strange that the `ui`  submodule is depending on the `UserProxyAgent`.